tensorflow/python/ops/image_grad.py

```diff
@@ -153,3 +153,40 @@ def _CropAndResizeGrad(op, grad):
       grad, op.inputs[0], op.inputs[1], op.inputs[2])
 
   return [grad0, grad1, None, None]
+
+@ops.RegisterGradient("ROIAlign")
+def _ROIAlignGrad(op, grad):
```

Comment:
Can you define a second derivative, even if it's slow and has to recompute things? It's important for TF's ecosystem to have a closed set of operations under differentiation.

Reply:
I haven't seen any second derivative examples in the codebase so I don't know how they are added to the framework. Could you please point me to some examples for second derivatives?

Reply:
Just RegisterGradient("ROIAlignGrad")

Reply:
I can try to do that. But when I tried to see examples of it, I saw only 22 out of 338 gradients have second derivatives. It seems that second derivatives are not that essential, possibly not even used. Do you still want second derivatives since this will involve adding new kernels increasing likelihood of potential problems.

Reply:
If the second derivative means another kernel then no, I don't think it's worth adding one, but I think it'd be nice to have one if it's expressible in terms of standard kernels.

Reply:
I don't think second derivative would be trivial.  Even if it was, I don't know how useful it would be for this op.

Reply:
Just for the future record a reason someone might want a second derivative for this is to optimize this with a higher order optimizer like kfac, which can scale to gigantic batch sizes, but requires fisher-vector products, which are like hessian-vector products, and whose efficient implementation would require a second derivative.

I don't know anyone wanting to do this now but I'm not willing to bet no one will in five years.